src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralResolutionService.java

```diff
@@ -78,22 +112,33 @@ public void processUpdatedRetentionCheck(final RetentionCheck retentionCheck) {
         // Ensure no race condition when we check other retention check statuses:
         referralResolutionRepository.lock(referralResolution, PESSIMISTIC_WRITE);
 
-        final var resolutionStatus = findResolution(referralResolution.getRetentionChecks());
+        final var resolutionStatus = findResolution(referralResolution.getRetentionChecks(), referralResolution.isProvisionalDeletionPreviouslyGranted());
 
         log.info("Updating offender referral '{}' to resolution status : '{}'",
                 referralResolution.getOffenderNumber(), resolutionStatus);
 
+        if(PROVISIONAL_DELETION_GRANTED == resolutionStatus){
+            referralResolution.setProvisionalDeletionPreviouslyGranted(true);
+        }
         referralResolution.setResolutionStatus(resolutionStatus);
         referralResolution.setResolutionDateTime(timeSource.nowAsLocalDateTime());
         referralResolutionRepository.save(referralResolution);
 
-        if (resolutionStatus == DELETION_GRANTED) {
+        if (DELETION_GRANTED == resolutionStatus) {
             deletionService.grantDeletion(referralResolution.getOffenderDeletionReferral());
         }
     }
 
+    public void retainReferral(OffenderDeletionReferral referral) {
+
+        log.info("Updating offender referral '{}' to resolution status : '{}'",
+            referral.getOffenderNumber(), RETAINED);
+
+        persistUpdatedReferral(referral, Collections.emptyList(), RETAINED);
```

Comment:
I wonder if it might be worth having another status to help explain why it was retained.  At this point, an offender that is retained because something has changed in the review duration would just be marked as retained but all of the retention checks would indicate they should be deleted? Or is there another way in the db that we'd be able to distinguish this?

Reply:
Good point, those offenders in theory will not be eligible for deletion for at least 7 years so it would make sense to distinguish them.